### PR TITLE
Functional tests - Contact bulk actions

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Test contact Bulk actions
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/shopParameters/contact/03_contactBulkActions.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17049)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]
Travis CI is :red_circle: but code is :ok_hand: for me.

File: tests/puppeteer/campaigns/functional/BO/shopParameters/contact/03_contactBulkActions.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      it('should create contact and check result', async function () {
+        const textResult = await this.pageObjects.addContactPage.createEditContact(test.args.contactToCreate);
+        await expect(textResult).to.equal(this.pageObjects.contactsPage.successfulCreationMessage);
+        const numberOfContactsAfterCreation = await this.pageObjects.contactsPage.getNumberOfElementInGrid();
+        await expect(numberOfContactsAfterCreation).to.be.equal(numberOfContacts + index + 1);
+      });
+    });
+  });
+
+  // 3 : Delete Contacts created with bulk actions
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/shopParameters/contact/03_contactBulkActions.js]:
```suggestion
  // 2 : Delete Contacts created with bulk actions
```

File: tests/puppeteer/campaigns/functional/BO/shopParameters/contact/03_contactBulkActions.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+const DashboardPage = require('@pages/BO/dashboard');
+const ContactsPage = require('@pages/BO/shopParameters/contact');
+const AddContactPage = require('@pages/BO/shopParameters/contact/add');
+// Importing data
+const ContactFaker = require('@data/faker/contact');
+
+let browser;
+let page;
+let numberOfContacts = 0;
+const firstContactData = new ContactFaker({description: 'todelete'});
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/shopParameters/contact/03_contactBulkActions.js]:
```suggestion
const firstContactData = new ContactFaker({title: 'todelete'});
```
maybe doing it with title is better

File: tests/puppeteer/pages/BO/shopParameters/contact/index.js [hunk 3]
```diff
@@ -120,4 +124,26 @@ module.exports = class Contacts extends BOBasePage {
...
+      this.page.click(this.selectAllRowsLabel),
+      this.page.waitForSelector(`${this.selectAllRowsLabel}:not([disabled])`, {visible: true}),
+    ]);
+    // Click on Button Bulk actions
+    await Promise.all([
+      this.page.click(this.bulkActionsToggleButton),
+      this.page.waitForSelector(this.bulkActionsToggleButton, {visible: true}),
+    ]);
+    // Click on delete and wait for modal
+    this.clickAndWaitForNavigation(this.bulkActionsDeleteButton);
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/contact/index.js]:
```suggestion
    await this.clickAndWaitForNavigation(this.bulkActionsDeleteButton);
```

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**